tools/swift-format/Sources/Rules/NoBlockComments.swift

```diff
@@ -14,5 +14,116 @@ import SwiftSyntax
+    // unless the comment is between tokens on the same line.
+    for piece in token.leadingTrivia {
+      if case .blockComment(let text) = piece,
+        !commentIsBetweenCode(token){
+        diagnose(.avoidBlockComment, on: token)
+        hasBlockComment = true
+        let lineCommentText = convertBlockCommentsToLineComments(text)
+        let lineComment = TriviaPiece.lineComment(lineCommentText)
+        pieces.append(lineComment)
+      }
+      else {
+        pieces.append(piece)
+      }
+    }
+    validToken =  validToken.withLeadingTrivia(Trivia.init(pieces: pieces))
+    return hasBlockComment ? validToken : token
+  }
+  
+  /// Returns a Boolean value indicating if the given trivia has a piece trivia
+  /// of block comment inline with code.
+  private func containsBlockCommentInline (trivia: Trivia) -> Bool {
+    // When the comment isn't inline with code, it doesn't need to
+    // to check that there are two spaces before the line comment.
+    if let firstPiece = trivia.first {
+      if case .newlines(_) = firstPiece {
+        return false
+      }
+    }
+    for piece in trivia {
+      if case .blockComment(_) = piece {
+        return true
+      }
+    }
+    return false
+  }
+  
+  /// Indicates if a block comment is between tokens on the same line.
+  /// If it does, it should only raise a lint error.
+  private func commentIsBetweenCode(_ token: TokenSyntax) -> Bool {
+    let hasCommentBetweenCode = token.leadingTrivia.isBetweenTokens
+    if hasCommentBetweenCode {
+      diagnose(.avoidBlockCommentBetweenCode, on: token)
+    }
+    return hasCommentBetweenCode
+  }
+  
+  /// Ensures there is always at least 2 spaces before the comment.
+  private func addSpacesBeforeComment(_ token: TokenSyntax) -> TokenSyntax {
+    let numSpaces = token.trailingTrivia.numberOfSpaces
+    if numSpaces < 2 {
+      let addSpaces = 2 - numSpaces
+      return token.withTrailingTrivia(token.trailingTrivia
+        .appending(.spaces(addSpaces)))
+    }
+    return token
+  }
+  
+  /// Receives the text of a Block comment and converts it to a Line Comment format text.
+  private func convertBlockCommentsToLineComments(_ text: String) -> String {
+    // Removes the '/*', '*/', the extra spaces and newlines from the comment.
+    let textTrim = text.dropFirst(2).dropLast(2)
+        .trimmingCharacters(in: .whitespacesAndNewlines)
+    
+    let splitComment = textTrim.split(separator: "\n", omittingEmptySubsequences: false)
+    var lineCommentText = [String]()
+    
+    for line in splitComment {
+      let startsComment = line.starts(with: " ") || line.count == 0 ? "//" : "// "
+      lineCommentText.append(startsComment + line)
+    }
+    return lineCommentText.joined(separator: "\n")
+  }
+}
+
+extension Diagnostic.Message {
+  static let avoidBlockComment = Diagnostic.Message(.warning, "Block comments should be avoided in favor of line comments.")
```

Comment:
Diagnostics should suggest the action the user should take to correct the violation. This should be something like `replace block comment with line comments`.

The one below should be `remove block comment inline with code` or something like that.